src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs

```diff
@@ -90,12 +96,12 @@ internal struct DeclaredSymbolInfo
             TextSpan span,
             ImmutableArray<string> inheritanceNames,
             int parameterCount = 0, int typeParameterCount = 0)
-            : this()
         {
-            Name = name;
-            NameSuffix = nameSuffix;
-            ContainerDisplayName = containerDisplayName;
-            FullyQualifiedContainerName = fullyQualifiedContainerName;
+            var stringTable = s_projectStringTable.GetOrCreateValue(project);
```

Comment:
So, technically, this should rarely be concurrent.  The incremental analyzer infrastructure will call into us to update these files one at a time.  But we can get concurrency if we're background computing, and someone explicitly tries to get an index.

The WeakRef approach is def interesting.  However, if GCs are frequent, i think we might end up in a situation where we literally GC this table between each file, thus negating the benefit of it.